app/routes/members-area.my-account/route.tsx

```diff
@@ -1,22 +1,193 @@
 import { LoaderFunctionArgs, redirect } from '@remix-run/node';
 import type { MetaFunction } from '@remix-run/react';
+import { useLoaderData } from '@remix-run/react';
+import classNames from 'classnames';
+import { useState } from 'react';
+import { useEcomApi } from '~/lib/ecom';
 import { initializeEcomApiForRequest } from '~/lib/ecom/session';
 
 import styles from './route.module.scss';
+import { Spinner } from '~/src/components/spinner/spinner';
+import { Dialog, DialogDescription, DialogTitle } from '~/src/components/dialog/dialog';
 
 export async function loader({ request }: LoaderFunctionArgs) {
     const api = await initializeEcomApiForRequest(request);
     if (!api.isLoggedIn()) {
         return redirect('/login');
     }
 
-    return null;
+    const user = await api.getUser();
+    return { user };
 }
 
 export default function MyAccountPage() {
+    const { user } = useLoaderData<typeof loader>();
+
+    const originalFirstName = user?.contact?.firstName ?? '';
```

Comment:
let's move everything to the hook that could be shared among all templates

Reply:
`useAccountDetails` or `useAccountDetailsForm`

Reply:
refactored